lib/rubocop/cop/mixin/autocorrect_unless_changing_ast.rb

```diff
@@ -28,6 +33,26 @@ def rewrite_node(node)
         c = correction(ps.ast)
         Corrector.new(ps.buffer, [c]).rewrite
       end
+
+      # 'begin' nodes with a single child can be removed without changing
+      # the semantics of an AST. Canonicalizing an AST in this way can help
+      # us determine whether it has really changed in a meaningful way, or
+      # not. This means we can auto-correct in cases where we would otherwise
+      # refrain from doing so.
+      #
+      # If any other simplifications can be done to an AST without changing
+      # its meaning, they should be added here (and the class renamed).
+      # This will make autocorrection more powerful across the board.
+      #
+      # @api private
```

Comment:
I'm not sure if we need these `@api` comments. We've only added them in
* lib/rubocop/cop/severity.rb
* lib/rubocop/cop/offense.rb
* lib/rubocop/formatter/base_formatter.rb

so far, and that's to support custom formatters. I mean, we need a well defined stable API for custom formatters, but formalizing other APIs can wait until we make a real plug-in architecture for custom cops.